### ISSUE TITLE
Closing the smiley chooser is difficult and very annoying

### ISSUE THREAD

**Issue by dumblob** [deltachat/deltachat-desktop#2818]
- Operating System (Linux/Mac/Windows/iOS/Android): Linux
- Delta Chat Version: Version 1.29.1 (git: v1.29.0-35-g14743055
- Expected behavior: Choosing a smiley immediately closes the smiley chooser. Pressing and holding Ctrl or Shift while clicking on smileys selects more smileys at once. De-pressing the Ctrl/Shift then inserts them and immediately closes the smiley chooser. Of course if one has focused/selected the search bar then no such closing shall happen and instead pressing Esc would close the chooser.
In addition to this (or as an alternative implementing an "avoidance scenario") a GitHub-like inline smiley prompt after pressing `:` could appear with a good fuzzy search (like GitHub) helping to construct a smiley from plain text.
- Actual behavior: One can close the smiley chooser only by clicking somewhere outside of the smiley chooser which is annoying because this way one loses the mouse cursor position prepared for another smiley :wink: (fortunately one does not lose the text cursor position).
- Steps to reproduce the problem: see "Expected behavior"
- Screenshots: N/A
- Logs: N/A

Basically the current behavior forces one to not use smileys at all :cry:.

Edit: removed the "loses text cursor position" rant - sorry about that detail, I did not test it properly.

**Comment by Simon-Laux** [comment]
can you make a small video showing this? I can not reproduce it

**Comment by r10s** [comment]
> Expected behavior: Choosing a smiley immediately closes the smiley chooser.

i do not know what is better, however, it is also not that clear wrt expectations: eg. Telegram does not close the selector as well, Signal closes it.

**Comment by dumblob** [comment]
True, it is not that clear wrt expectations.

Let me re-iterate that I speak here only about the desktop version which is expected to work without hassle with traditional equipment - namely a physical keyboard and physical mouse (or touchpad). And clearly the current behavior is far from "without hassle" :wink:.

My problem is really the huge annoyance. I actually do not care how this will be tackled - the ideas I sketched above are just ideas, not my expectations. My only expectation is that I will not need to do **so much of focused work** only to put one smiley in the text I am currently writing.

**Comment by obsoleszenz** [comment]
I cannot reproduce your situation, yes the emoji picker only closes if you click outside, but the focus in the text message is not getting lost, i can immediatly continue writing when the emoji picker is closed (also while it's still open). Don't really understand your problem. Which DE are you using? Probably electron behaves weirdly there?

**Comment by dumblob** [comment]
Thanks for pointing out I actually do not have to position the mouse cursor at the text cursor but write further instead!

That is undoubtedly a viable option but due to the oscured conversation (i.e. overlapped by the smiley chooser) this is in its current form not an option for me.

Therefore another simple idea - maybe typing further characters (except for `Tab` and `Shift+Tab` which should be used to navigate inside of the smiley chooser) could automatically close the smiley chooser?

This should be easy to implement and would solve this issue (at least for me). I would then maybe propose in the DC forum a feature request to implement inline smiley suggestions (after typing a standalone `:` character) like GitHub does - but that is not a topic of this issue here.

I have edited my initial post above to clarify this point.

**Comment by Jikstra** [comment]
Can you post screenshots or something? Maybe I'm too used to the telegram ux but to me the emoji chooser is not really a problem, or at least not unusable as to describe it :D But we devs are biased ^^

**Comment by Simon-Laux** [comment]
about closing the picker after clicking on an emoji:
element keeps it open, discord closes it.

Also btw: if you remember the right code you can already type `:colon_emojis:`, though there is no autocompletion yet.

**Comment by dumblob** [comment]
Yeah, autocompletion (with "meta" fuzzy matching - writing `:tear` would offer a smiley with code `:cry:` as GitHub does) would solve it for my use cases :wink:.

**Comment by dumblob** [comment]
Now I noticed I did not manage to post a screenshot and am sorry for that. Here we go:

![20230201_094049+0100-1680x1050+0+0-imlib2_grab](https://user-images.githubusercontent.com/2451044/215995439-c0ab0eb1-b46d-4996-9681-e19e1ca471f3.png)

**Comment by Simon-Laux** [comment]
this should be solved in newer versions >=1.34.4
